### ISSUE TITLE
`Print Assumptions` does not report local `-type-in-type` setting

### ISSUE THREAD

**Issue by maximedenes** [coq/coq#7692]
This is close to a critical bug.

If `test.v` is:

```coq
Require Import Hurkens.
Definition bad := PropNeqType.paradox eq_refl.
```

and I run:
```
coqc -type-in-type test.v
coqtop
```

The following script:
```coq
Require Import test.
Print Assumptions test.bad.
```

prints:
```
Axioms:
bad : False
```

This is a serious design mistake that probably dates back to when local support for `-type-in-type` was added (cc @ppedrot).

`Print` reports correctly that the definition depends on `-type-in-type`.

**Comment by ppedrot** [comment]
Strange, I was pretty sure I implemented (at least partially) this check. I'll have a look.

**Comment by maximedenes** [comment]
It would be good to also make the documentation of this flag better. In particular, please clarify what is the expected behavior w.r.t. separate compilation. With the current doc, I don't even know if this is a `coqtop` bug or a `Print Assumptions` bug.

**Comment by ppedrot** [comment]
Why would it be a coqtop bug? Do you mean that this flag could be contaminating in some sense?

**Comment by maximedenes** [comment]
Yes. If the flag is meant to be "contaminating" (it used to be), then it is a `coqtop` bug (a `library` bug, more precisely). If not, it is simply `Print Assumptions` not picking the right flag (global instead of local). The documentation should make clear what is the expected behavior.

**Comment by ppedrot** [comment]
That's very strange, the assumption code for constants does add them when they have the type-in-type flag set (that's not the case for inductive types and constructors, which is a separate bug).

**Comment by ppedrot** [comment]
@maximedenes Looking back at the example, what is wrong with the output? It's true that `bad` is an axiom in the sense that it relies on type-in-type. The code gathering assumptions rightfully adds this definition to the set of axioms as it uses a collapsed hierarchy. I agree that the printer could be more informative, but it's not unsound in any way.

We can indeed rework assumptions to gather more information, but by no means the current behaviour is a critical bug.

**Comment by maximedenes** [comment]
Ah, I see, I hadn't realized this was the intended behavior. So indeed, a more informative printer like "`bad` relies on `-type-in-type`" would be helpful.

**Comment by SkySkimmer** [comment]
now says
~~~
Axioms:
bad relies on an unsafe hierarchy.
~~~
